salt/modules/ssh.py

```diff
@@ -222,6 +222,23 @@ def _validate_keys(key_file, fingerprint_hash_type):
 
     return ret
 
+def _convert_fingerprint_format(fingerprint):
+    try:
+        if fingerprint and fingerprint.find(':') == -1:
+            fingerprint_len = int(4 * ((len(fingerprint) + 3) / 4))
```

Comment:
It'll not work properly on Py3 because it will do `4* ... /4` as a floating point operation. Use integer division operator instead `//`:
```python
fingerprint_len = 4 * ((len(fingerprint) + 3) // 4)
# or a bit more readable (less brackets):
fingerprint_len = (len(fingerprint) + 3) // 4 * 4
```